src/armlsd.py

```diff
@@ -60,6 +60,7 @@ def ARMLSD(argflag, spect, fitsdict, reuseMaster=False):
         slf = sciexp[sc]
         scidx = slf._idx_sci[0]
         msgs.info("Reducing file {0:s}, target {1:s}".format(fitsdict['filename'][scidx], slf._target_name))
+        msgs.sciexp = slf  # For QA writing on exit, if nothing else.  Could write Masters too
```

Comment:
I fear it is necessary to close the file.  We are using
qa.savefig() in our QA routines to generate the pages.
But a pp.close() is needed to wrap up the PDF file.

And indeed, this is only one step towards the QA fix.
Is it possible to get sciexp in the traceback when PYPIT
crashes?


> On Apr 19, 2016, at 8:55 AM, Ryan Cooke <notifications@github.com> wrote:
> 
> In src/armlsd.py <https://github.com/PYPIT/PYPIT/pull/43#discussion_r60258460>:
> 
> > @@ -60,6 +60,7 @@ def ARMLSD(argflag, spect, fitsdict, reuseMaster=False):
> >          slf = sciexp[sc]
> >          scidx = slf._idx_sci[0]
> >          msgs.info("Reducing file {0:s}, target {1:s}".format(fitsdict['filename'][scidx], slf._target_name))
> > +        msgs.sciexp = slf  # For QA writing on exit, if nothing else.  Could write Masters too
> I'm happy with your approach, but it's probably worthwhile having a brief discussion about this.
> 
> If a bug is encountered (and not an error message), then the QA will not be dumped. The only way this can be done is to throw a general try/except clause around ARMLSD, which I hesitate to do (because there's already a try/except clause enclosing the call to PYPIT).
> 
> It seems we can just include slf.sciexp._qa.savefig() whenever we want a page written to be written to file. See: this link <http://matplotlib.org/examples/pylab_examples/multipage_pdf.html>
> I've not tried this, so I'm not sure if it will work.
> 
> Another option might be to remove the try/except around PYPIT, and instead put it within ARMLSD, specifically:
> 
> def ARMLSD(...):
>     try:
>         reduce data
>     except:
>         slf.sciexp._qa.close()
>         then, print traceback, as in the __init__ function in pypit.py
> which, again, seems like a dirty answer to our problem.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/PYPIT/PYPIT/pull/43/files/ce9c3eeb928cf7ad173485d0e4fbab1a901d7798#r60258460>
> !DSPAM:826,571654e7145431444819054!

----------------------------------------------
J. Xavier Prochaska
UCO/Lick Observatory
1156 High St.
UC Santa Cruz
Santa Cruz, CA 95064

http://www.ucolick.org/~xavier/
http://imps.ucolick.org/
http://www.qpqsurvey.org/
http://k1dm3.ucolick.org/
http://www.ucolick.org/~xavier/IDL/index.html
https://github.com/pyigm/pyigm
https://github.com/linetools/linetools

831-459-2135 (Direct)
831-459-2991 (UCO/Lick Main)
831-459-5244 (Fax)